### PR TITLE
Support google provider v5xx - update data-type for node taint

### PR DESCRIPTION
**Pull request by siddharth1904**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -224,8 +224,17 @@ resource "google_container_node_pool" "node_pools" {
     service_account = module.gke_service_account.email
     oauth_scopes    = local.oauth_scopes
     tags            = distinct(concat(local.default_network_tags, each.value.network_tags))
-    taint           = each.value.node_taints
-    metadata        = each.value.node_metadatas #see: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#metadata
+    dynamic "taint" {
+      for_each = each.value.node_taints
+      iterator = node_taint
+
```
Comment (nurairasia) [inline main.tf:230]: this new line isnt needed right ?

Comment (siddharth1904) [reply main.tf]: @nurairasia removed the newline. Resolving this thread

**Review by nurairasia — APPROVED** [review]
LGTM

**Comment by prajithparammal** [comment]
LGTM

**Review by prajithparammalairasia — APPROVED** [review]
LGTM

**Pull request merged**